tests/integration/rosetta/rosetta_test.go

```diff
@@ -0,0 +1,272 @@
+			wantStatusCode: http.StatusUnprocessableEntity,
+			wantHandlerErr: assert.Error,
+		},
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			// prepare request payload (JSON)
+			enc, err := json.Marshal(test.request)
+			require.NoError(t, err)
+
+			// create request
+			req := httptest.NewRequest(http.MethodPost, "/account/balance", bytes.NewReader(enc))
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			rec := httptest.NewRecorder()
+
+			ctx := echo.New().NewContext(req, rec)
+
+			// execute the request
+			err = rosettaSvc.Balance(ctx)
+			test.wantHandlerErr(t, err)
+
+			// validate response data
+
+			// validate status code
+			if test.wantStatusCode != http.StatusOK {
+
+				// ugly workaround for HTTP status code set by router vs set on context
+				e, ok := err.(*echo.HTTPError)
+				require.True(t, ok)
+				assert.Equal(t, test.wantStatusCode, e.Code)
+
+				// nothing more to do, response validation should only be done for '200 OK' responses
+				return
+			}
+
+			assert.Equal(t, test.wantStatusCode, rec.Result().StatusCode)
+
+			// unpack response
+			var balanceResponse rosetta.BalanceResponse
+			require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &balanceResponse))
+
+			// verify that the block data matches the input data
+			assert.Equal(t, test.request.BlockID.Index, balanceResponse.BlockID.Index)
+			assert.Equal(t, test.request.BlockID.Hash, balanceResponse.BlockID.Hash)
+
+			// verify that we have precisely one balance in the response
+			if assert.Len(t, balanceResponse.Balances, 1) {
+
+				// verify the balance data - both the value and that the output matches the input spec
+				balance := balanceResponse.Balances[0]
+				assert.Equal(t, test.request.Currencies[0].Symbol, balance.Currency.Symbol)
+				assert.Equal(t, test.request.Currencies[0].Decimals, balance.Currency.Decimals)
+				assert.Equal(t, test.wantBalance, balance.Value)
+			}
+		})
+	}
+}
+
+// TestGetBalanceBadRequest tests whether an improper JSON (e.g. wrong field types) will cause a '400 Bad Request' error
+func TestGetBalanceBadRequest(t *testing.T) {
+
+	// JSON with an invalid structure (integer instead of string for network name)
+	payload := `{ "network_identifier": { "blockchain": "flow", "network": 99} }`
+
+	// create request
+	req := httptest.NewRequest(http.MethodPost, "/account/balance", strings.NewReader(payload))
+	req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+	rec := httptest.NewRecorder()
+
+	ctx := echo.New().NewContext(req, rec)
+
+	// execute the request
+	err := rosettaSvc.Balance(ctx)
+	assert.Error(t, err)
+
+	e, ok := err.(*echo.HTTPError)
+	require.True(t, ok)
+	assert.Equal(t, http.StatusBadRequest, e.Code)
+}
+
+// getBalanceRequest will generate a BalanceRequest with the specified parameters.
+func getBalanceRequest(address string, blockIndex uint64, blockHash string) rosetta.BalanceRequest {
```

Comment:
I'd suggest renaming this to `balanceRequest`, since `get` implies that we're not generating it but getting it from somewhere, WDYT? The same applies to the other `getXXX` methods below